### PR TITLE
Add InstanceName for Mesh Requests

### LINKED ISSUE #9050: Add Validation for Bind-Mount API and Authorization Token for Mesh Service API's
#### What problem would the feature you're requesting solve? Please describe.
Received a MSRC vulnerability in our mesh service client's API's. 
_Azure Function App APIs allow lower privileged (Reader) users to read sensitive files from a Function App container’s file system, while not having direct access to the files from the Azure Portal. This issue is exploited by modifying an undocumented API call that is utilized by the Azure Portal to read Function App files._ 

#### Describe the solution you'd like
As defense in depth, we will add 2 measures 
1)Block the mount operation as soon as soon as we can identify the platform doesn't need the capability anymore (this will not be completely foolproof since there is a gap between the time, we can identify vs time we block) 
2)Add a authorization token to the API calls to the API server within the container. Note: If the attacker is able to gain access to the container, this authorization token would also be accessible to the attacker, but it would add an additional step for the attacker to perform."

### PR DESCRIPTION
**Pull request by Tonewall**
Adding InstanceName (ContainerName) for the requests sent to the mesh service. PR for mesh: https://msazure.visualstudio.com/One/_git/AAPT-Antares-Functions-Docker/pullrequest/7361854

### Issue describing the changes in this PR

resolves #9050

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Comment by balag0** [comment]
Thanks, Would be good to get the server side changes finalized before coming back to this..

File: src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs [hunk 1]
```diff
@@ -142,8 +145,19 @@ private async Task<HttpResponseMessage> SendAsync(IEnumerable<KeyValuePair<strin
         {
             var operationName = formData.FirstOrDefault(f => string.Equals(f.Key, Operation)).Value;
             _logger.LogDebug($"Sending mesh request {operationName}");
-            var res = await _client.PostAsync(_environment.GetEnvironmentVariable(EnvironmentSettingNames.MeshInitURI),
-                new FormUrlEncodedContent(formData));
+            //var res = await _client.PostAsync(_environment.GetEnvironmentVariable(EnvironmentSettingNames.MeshInitURI),
+            //    new FormUrlEncodedContent(formData));
+            //var res = await _client.PostAsync(string requestUri, HttpContent content);
```
Comment (kshyju) [inline src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs]: Remove commented out code.

File: src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs [hunk 2]
```diff
@@ -159,5 +173,17 @@ private static string Serialize<T>(T o)
 
             return serialized;
         }
+
+        private KeyValuePair<string, string> GetInternalRequestBody()
```
Comment (kshyju) [inline src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs]: Is this method unused?

File: src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs [hunk 3]
```diff
@@ -142,8 +145,19 @@ private async Task<HttpResponseMessage> SendAsync(IEnumerable<KeyValuePair<strin
...
+            //var res = await _client.PostAsync(_environment.GetEnvironmentVariable(EnvironmentSettingNames.MeshInitURI),
+            //    new FormUrlEncodedContent(formData));
+            //var res = await _client.PostAsync(string requestUri, HttpContent content);
+
+            var request = new HttpRequestMessage(HttpMethod.Post, _environment.GetEnvironmentVariable(EnvironmentSettingNames.MeshInitURI))
+            {
+                Content = new FormUrlEncodedContent(formData)
+            };
+
+            request.Headers.Add("fx-current-instance", _environment.GetEnvironmentVariable(EnvironmentSettingNames.ContainerName));
```
Comment (kshyju) [inline src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs]: I see we have a[ constant file for some other headers](https://github.com/Azure/azure-functions-host/blob/afafa87da101d8ed2fe1fa7aee317692db252786/src/WebJobs.Script/ScriptConstants.cs#L100-L107). Should we move this there? 

Also what is this "fx-" prefix stands for?

File: src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs [hunk 4]
```diff
@@ -142,8 +145,19 @@ private async Task<HttpResponseMessage> SendAsync(IEnumerable<KeyValuePair<strin
...
+
+            var res = await _client.SendAsync(request);
+
             _logger.LogDebug($"Mesh response {res.StatusCode}");
             return res;
```
Comment (kshyju) [inline src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs:158]: nit: Add a newline before return.

File: src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs [hunk 5]
```diff
@@ -6,6 +6,8 @@
 using System.IO;
 using System.Linq;
 using System.Net.Http;
+using System.Net.Http.Formatting;
```
Comment (kshyju) [inline src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs]: Unused ? In Visual studio, right click here and select "Remove and Sort usings" from the context menu.

**Comment by kshyju** [comment]
> Thanks, Would be good to get the server side changes finalized before coming back to this..

@Tonewall Is this PR blocked?

File: src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs (hunk 3 above)
Comment (Tonewall) [reply src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs]: It's just a prefix for Functions we're reserving for headers for the linux consumption's mesh process. We can change it to whatever

File: src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs (hunk 2 above)
Comment (Tonewall) [reply src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs]: Yes, I forgot to remove it when changing from body to headers for the request, will remove it!

**Review by liliankasem — APPROVED** [review]

File: src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs (hunk 5 above)
Comment (Tonewall) [reply src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs]: Fixed!

File: src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs (hunk 3 above)
Comment (Tonewall) [reply src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs]: Moved the header to ScriptConstants 👍

File: src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs (hunk 2 above)
Comment (Tonewall) [reply src/WebJobs.Script.WebHost/Management/MeshServiceClient.cs]: Removed!

File: src/WebJobs.Script.WebHost/Models/InternalRequestBody.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (balag0) [inline src/WebJobs.Script.WebHost/Models/InternalRequestBody.cs]: this is still needed?

Comment (Tonewall) [reply src/WebJobs.Script.WebHost/Models/InternalRequestBody.cs]: Removed, thank you for that catch

**Review by balag0 — APPROVED** [review]

**Pull request merged**